### DISCUSSION TITLE
Solution to keep atomic when uploading files and updating DB in one transaction

### DISCUSSION THREAD

**Discussion by hereAlexT** [supabase/supabase#19895] [Questions]
In some cases, like posting, pics need to be sent along with DB records. 
I want to keep the atomic and consistency between database records and file storage.

I have thought up some solutions but I just feel that they are not the most elegant way.
1. **Client-side transaction-like method:** 
One drawback of this is that when users give up the post while uploading, need a server-side cleanup to clean orphan pics.
2. **Edge function**
Create an edge function to handle pic uploading and DB inserting, the function also needs to have roll-back mechanism.

I am thinking is there any better solution on supabase? And what's the most common solution there?

**Comment by MentalGear** [comment]
I'm also in the same situation and I find the supabase docs lacking. While of really good detail otherwise, it doesn't detail on how to ensure ACID transactions for storage / db record pairs (@soedirgo knows more?).

I was thinking of a 3rd option: a function that triggers whenever a new file is added on a **supabase storage** bucket. But searching the discussions, I'm wondering if this is even [supported reliably](https://github.com/orgs/supabase/discussions/19017#discussioncomment-8110062) on supabase.

EDIT: Apparently such triggers can be added to the default storage table (described [here](https://github.com/orgs/supabase/discussions/19017#discussioncomment-7613197)).
